splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java

```diff
@@ -837,13 +826,41 @@ public static ExecRow generateOutputRow(String schemaName, String tableName, Exe
     }
 
 
-    public static Iterable displayTableStatistics(List<StatisticsOperation> futures, boolean mergeStats, final DataDictionary dataDictionary, final TransactionController tc, final HashMap<Long,Pair<String,String>> displayPair) {
+    public static Iterable displayTableStatistics(ArrayList<StatisticsOperation> collectOps,
+                                                  boolean mergeStats,
+                                                  final DataDictionary dataDictionary,
+                                                  final TransactionController tc,
+                                                  final HashMap<Long, Pair<String, String>> displayPair) throws StandardException {
+        // Schedule the first <maximumConcurrent> jobs
+        int maximumConcurrent = EngineDriver.driver().getConfiguration().getCollectSchemaStatisticsMaximumConcurrent();
+        for (int i = 0; i < maximumConcurrent && i < collectOps.size(); ++i) {
+            collectOps.get(i).openCore();
+        }
+        // Handle the next jobs as we go: (One job returns -> one job can start), ensuring <maximumConcurrent> jobs at all time
+        Iterable<StatisticsOperation> movingExecutionWindow = () -> new Iterator<StatisticsOperation>() {
+            int i = 0;
+            @Override
+            public boolean hasNext() {
+                return i < collectOps.size();
+            }
+
+            @Override
+            public StatisticsOperation next() {
+                if ((long)i + (long)maximumConcurrent < (long)collectOps.size()) {
```

Comment:
Why are we casting to long? If it's to avoid integer overflow we'd have it anyway on collectOps.get(i + maximumConcurrent)

Reply:
I disagree, collectOps.size() returns an int, so if we go in the loop, we know that i+maximumConcurrent does not overflow.
If i+maximumConcurrent were to overflow line 849, casting would prevent the overflow and we would not enter the if.

Reply:
Oh, and by the way, if someone sets maximumConcurrent to MAXINT, we **will** overflow.

Reply:
I see, that makes sense, thanks!